scripts/sensor/_02_decoding_full_epochs.py

```diff
@@ -0,0 +1,202 @@
+
+
+@failsafe_run(on_error=on_error, script_path=__file__)
+def run_epochs_decoding(*, cfg, subject, condition1, condition2, session=None):
+    msg = f'Contrasting conditions: {condition1} – {condition2}'
+    logger.info(**gen_log_kwargs(message=msg, subject=subject,
+                                 session=session))
+
+    fname_epochs = BIDSPath(subject=subject,
+                            session=session,
+                            task=cfg.task,
+                            acquisition=cfg.acq,
+                            run=None,
+                            recording=cfg.rec,
+                            space=cfg.space,
+                            suffix='epo',
+                            extension='.fif',
+                            datatype=cfg.datatype,
+                            root=cfg.deriv_root,
+                            check=False)
+
+    epochs = mne.read_epochs(fname_epochs)
+    if cfg.analyze_channels:
+        # We special-case the average reference here to work around a situation
+        # where e.g. `analyze_channels` might contain only a single channel:
+        # `concatenate_epochs` below will then fail when trying to create /
+        # apply the projection. We can avoid this by removing an existing
+        # average reference projection here, and applying the average reference
+        # directly – without going through a projector.
+        if 'eeg' in cfg.ch_types and cfg.eeg_reference == 'average':
+            epochs.set_eeg_reference('average')
+        else:
+            epochs.apply_proj()
+        epochs.pick(cfg.analyze_channels)
+
+    # We define the epochs and the labels
+    if isinstance(cfg.conditions, dict):
+        epochs_conds = [cfg.conditions[condition1],
+                        cfg.conditions[condition2]]
+        cond_names = [condition1, condition2]
+    else:
+        epochs_conds = cond_names = [condition1, condition2]
+        epochs_conds = [condition1, condition2]
+
+    # We have to use this approach because the conditions could be based on
+    # metadata selection, so simply using epochs[conds[0], conds[1]] would
+    # not work.
+    epochs = mne.concatenate_epochs([epochs[epochs_conds[0]],
+                                     epochs[epochs_conds[1]]])
+
+    n_cond1 = len(epochs[epochs_conds[0]])
+    n_cond2 = len(epochs[epochs_conds[1]])
+
+    X = epochs.get_data()
+    y = np.r_[np.ones(n_cond1), np.zeros(n_cond2)]
+
+    with config.get_parallel_backend():
+        classification_pipeline = make_pipeline(
+            Scaler(scalings='mean'),
+            Vectorizer(),  # So we can pass the data to scikit-learn
+            LogisticRegression(
+                solver='liblinear',  # much faster than the default
+                n_jobs=1
+            )
+        )
+
+        # Now, actually run the classification, and evaluate it via a
+        # cross-validation procedure.
+        cv = StratifiedKFold(shuffle=True, n_splits=cfg.decoding_n_splits)
```

Comment:
Using shuffle without fixing the seed is dangerous.

Reply:
Good catch!!! Simply forgot that one!